FinanceChargesListener/Gateway/Interfaces/IChargesApiGateway.cs

```diff
@@ -15,5 +16,6 @@ public interface IChargesApiGateway
         Task<Charge> GetById(Guid chargeId, Guid assetId);
         Task DeleteBatchAsync(List<ChargeKeys> chargeIds, int batchCapacity);
         Task<List<ChargeKeys>> GetChargesByYearGroupSubGroup(short chargeYear, ChargeGroup chargeGroup, ChargeSubGroup? chargeSubGroup);
+        Task<IList<Charge>> GetChargesAsync(PropertyChargesMessageSqs queryParameters);
```

Comment:
You have duplicated the method, may be you can modify the previous method to return charge and then after the response you can extract the charge keys from there.